src/adapter/vertex.rs

```diff
@@ -165,10 +173,17 @@ impl<'a> Vertex<'a> {
     }
 
     pub(super) fn as_variant(&self) -> Option<&'a Variant> {
-        self.as_item().and_then(|item| match &item.inner {
-            rustdoc_types::ItemEnum::Variant(v) => Some(v),
+        match &self.kind {
+            VertexKind::Variant(variant) => variant.variant(),
             _ => None,
-        })
+        }
+    }
+
+    pub(super) fn as_enum_variant(&self) -> Option<&'a EnumVariant> {
```

Comment:
This signature is incorrect, and is causing the borrowing/escaping error you were dealing with:
```suggestion
    pub(super) fn as_enum_variant(&self) -> Option<&'_ EnumVariant<'a>> {
```

We can't create a `&'a EnumVariant` from an owned `VertexKind::Variant(EnumVariant<'a>)`. We can only create a `&'_ EnumVariant<'a>` so that's what we should return. Note how everything that returns a `&'a T` here actually held a `&'a T` internally, whereas everything else (e.g. `Attribute`) returns a `&'_ T`.

As an additional note, it's generally a bad idea to elide lifetime arguments from the type, like the `&'a EnumVariant` here instead of `&'a EnumVariant<'a>`. If we had the latter here, the issue would have been much more obvious.

Reply:
Thanks for the catch; didn't even realize I elided!